irida_staramr_results/cli.py

```diff
@@ -0,0 +1,69 @@
+import sys
+import argparse
+import logging
+from version import __version__
+from irida_api import IridaAPI
+from urllib.error import HTTPError
+
+
+def init_argparser():
+    argument_parser = argparse.ArgumentParser(
+        prog="irida-staramr-results",
+        description="Exports StarAMR results available through IRIDA into a single excel report."
+    )
+    argument_parser.add_argument("-v", "--version", action="version", version=f"{argument_parser.prog} {__version__}",
+                                 help="The current version of irida-staramr-results.")
+    argument_parser.add_argument("-p", "--project", action="store",
+                                 help="Project(s) to scan for StarAMR results.")
+    argument_parser.add_argument("-o", "--output", action="store",
+                                 help="The name of the output excel file.")
+
+    # Arguments that will override fields in the config file.
+    argument_parser.add_argument("-u", "--username", action="store",
+                                 help="Override the 'username' in config file. This is your IRIDA account username.")
+    argument_parser.add_argument("-pw", "--password", action="store",
+                                 help="Override the 'password' in config file. This is your IRIDA account password.")
+
+    return argument_parser
+
+
+def main():
+    """
+    Main entry point of irida_staramr_results.
+    Accepts commands form terminal to be processed by the program.
+    """
+    argument_parser = init_argparser()
+
+    if len(sys.argv) == 1:
+        argument_parser.print_help()
+        sys.exit(1)
+
+    args = argument_parser.parse_args()
+    args_dict = vars(args)
+
+    if args_dict["project"] is None:
```

Comment:
Since the program does not work without a project, you can use logging.error instead of logging.warning, and then use a sys.exit(1)
This way you don't need all your core code in an else statement.

Reply:
Done.